### PR TITLE
Update to v10.10

### PR DESCRIPTION
**Pull request by lwbt**
Update resources and dependencies for v10.10 release.
Also update runtime from 23.08 to 24.08 and runtime extensions to the latest versions.

Please merge https://github.com/flathub/org.jellyfin.JellyfinServer/issues/359 before merging this branch to build v10.9 first and to better keep track of changes from 10.8.z -> 10.9.z -> 10.10.z.

### CODE REVIEW

File: org.jellyfin.JellyfinServer.yml [hunk 1]
```diff
@@ -21,16 +23,19 @@ cleanup:
   - /share/info
   - /share/man
   - /share/pkgconfig
+  - /share/vpl/examples
 
 finish-args:
   - --device=dri
   - --filesystem=home/Jellyfin Server Media:create
+  - --filesystem=home/.var/app/org.jellyfin.JellyfinServer/media:create
```
Comment (bbhtt) [inline org.jellyfin.JellyfinServer.yml:31]: this is not needed, the app has access to its own ~/.var/app/$FLATPAK_ID directory.

File: org.jellyfin.JellyfinServer.yml [hunk 2]
```diff
@@ -21,16 +23,19 @@ cleanup:
...
   - --filesystem=/media:ro
   - --filesystem=/mnt:ro
   - --filesystem=/run/media:ro
   - --share=network
   - --env=PATH=/app/extensions/bin:/app/bin:/usr/bin
   - --env=LD_LIBRARY_PATH=/app/extensions/lib:/app/lib
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline org.jellyfin.JellyfinServer.yml:38]: libnotify uses the portal from 0.8.3, not needed.

File: org.jellyfin.JellyfinServer.yml [hunk 3]
```diff
@@ -214,29 +227,20 @@ modules:
...
-          - for p in $(cat debs/patches/series); do if [[ $p != "#"* ]]; then patch
-            -Np1 -i "debs/patches/$p" -d .; fi; done
+          type: git
+          tag-pattern: ^v([\d.]+)$
 
   - name: libopenmpt
+    buildsystem: simple
+    build-commands:
+      #- ./configure
+      - make -j$FLATPAK_BUILDER_N_JOBS PREFIX=$FLATPAK_DEST install
```
Comment (bbhtt) [inline org.jellyfin.JellyfinServer.yml:243]: drop buildsystem and the rest, add `no-autogen: true`

File: org.jellyfin.JellyfinServer.yml [hunk 4]
```diff
@@ -457,144 +361,40 @@ modules:
...
+      # Already included as a submodule: glad, jinja, markupsafe
 
   - name: numactl
-    buildsystem: autotools
+    #buildsystem: autotools
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix $FLATPAK_DEST
+      - make -j$FLATPAK_BUILDER_N_JOBS PREFIX=$FLATPAK_DEST install
```
Comment (bbhtt) [inline org.jellyfin.JellyfinServer.yml:387]: this should use autotools buildsystem

File: org.jellyfin.JellyfinServer.yml [hunk 5]
```diff
@@ -605,27 +405,22 @@ modules:
...
+        commit: da14df5535fd46776fb1c9da3130973295c87aca
+        #tag: 0.164.r3194
+        # Every commit to the master branch is considered a release
+        # https://code.videolan.org/videolan/x264/-/issues/35
+        x-checker-data:
+          type: json
+          url: https://code.videolan.org/api/v4/projects/536/repository/commits
+          commit-query: first( .[].id )
+          version-query: first( .[].id )
+          timestamp-query: first( .[].committed_date )
```
Comment (bbhtt) [inline org.jellyfin.JellyfinServer.yml:423]:
```suggestion
```

x264, doesn't do releases, don't follow commits. They bump so version.

File: org.jellyfin.JellyfinServer.yml (hunk 3 above)
Comment (lwbt) [reply org.jellyfin.JellyfinServer.yml:243]: I get this error message when I try your suggested changes:

```
install: cannot create directory ‘/usr/local’: Read-only file system
```

File: org.jellyfin.JellyfinServer.yml (hunk 4 above)
Comment (lwbt) [reply org.jellyfin.JellyfinServer.yml:387]: I get this error message when I try your suggested changes on a local build:

```
configure: error: cannot find required auxiliary files: compile config.guess config.sub ltmain.sh missing install-sh
```

I can see that:

- Handbrake is using it in a different way: https://github.com/flathub/fr.handbrake.ghb/blob/master/fr.handbrake.ghb.json
- Sunshine uses an older version from Debian: https://github.com/flathub/dev.lizardbyte.app.Sunshine/blob/master/dev.lizardbyte.app.Sunshine.yml

I opt for the way Handbrake does it adding `rm-configure: true` leaving `#buildsystem: autotools` and dropping the other lines.

File: org.jellyfin.JellyfinServer.yml (hunk 3 above)
Comment (bbhtt) [reply org.jellyfin.JellyfinServer.yml:221]: Add PREFIX in `make-install-args`

**Comment by lwbt** [comment]
bot, build

File: org.jellyfin.JellyfinServer.yml (hunk 4 above)
Comment (bbhtt) [reply org.jellyfin.JellyfinServer.yml]: > Handbrake is using it in a different way: https://github.com/flathub/fr.handbrake.ghb/blob/master/fr.handbrake.ghb.json

this one is fine, if nothing is specified it defaults to autotools.

File: org.jellyfin.JellyfinServer.yml [hunk 6]
```diff
@@ -52,6 +57,26 @@ add-extensions:
...
+      - --buildtype=release
+      - --default-library=shared
+      - -Dtests=false
+      - -Dintrospection=disabled
+      - -Dman=false
+      - -Dgtk_doc=false
+      - -Ddocbook_docs=disabled
+    sources:
+      - type: git
+        url: https://salsa.debian.org/gnome-team/libnotify.git
```
[the remark this answers is not in the document]
Comment (lwbt) [reply org.jellyfin.JellyfinServer.yml]: Btw, I only see libnotify listed under Gnome here: https://docs.flatpak.org/en/latest/available-runtimes.html

 wouldn't have included it if the document said its included and I didn't expect it to be included in a newer version of the runtime. I'm certain I checked contents of an older runtime.

**Comment by lwbt** [comment]
bot, build

**Pull request merged**